test/unit_tests/braket/experimental/autoqasm/test_pulse_control.py

```diff
@@ -0,0 +1,166 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+"""Tests for the pulse control module."""
+
+import textwrap
+
+import pytest
+
+import braket.experimental.autoqasm as aq
+from braket.experimental.autoqasm.instructions import (
```

Comment:
Can we keep these separate? `aq.pulse` should contain all the pulse instructions

Reply:
I think how we structure the code generally needs some review. Since AutoQASM folders has quantum program parts like program, instructions, and also infrastructure parts like autograph, api, transpiler, operators etc. I think it make sense to bring the quantum part close together (ie., pulses, gates, measure under "instructions"). 

But I agree that bringing quantum parts closer to the top level is good because it's the parts that customers actually use. We may want to consider a module structure that hide all the infrastructure part, and expose only quantum part to the top level. Also, without the change of module structure, we can also alias useful parts to the top level.

Reply:
Also, I was planning to reserve `aq.pulse` for the decorator.

Reply:
In sum, you'd like to postpone making changes? When we do, we should have pulse and gates in different modules. The import path should convey they they operate in different contexts. This could be useful in a program:
```
@aq.function
def my_prog:
    inst.h(q0)
    pulse.set_frequency(frame0)
```

Reply:
About import path, pulse and gates are in different sub-module, so we can already do
```
from braket.experimental.autoqasm.instructions.gates import h, cnot
from braket.experimental.autoqasm.instructions.pulse_control import set_phase, shift_frequency 
```
I can make this changes in tests, and can stick to this convention when we create notebook

Reply:
We can already start to make the change on directory structure today, but want to get more feedback. When we do I'd like to hide the "operators" module from top level, and changed "instructions" to "operations", which contains gates , reset and measure. So in your example, it could be

```
@aq.function
def my_prog:
    ops.h(q0)
    pulse.set_frequency(frame0)
```